lib/bn.js

```diff
@@ -503,10 +503,14 @@
 
   BN.prototype.toBuffer = function toBuffer (endian, length) {
     assert(typeof Buffer !== 'undefined');
-    return new Buffer(this.toArray(endian, length));
+    return this.toArrayLike(Buffer, endian, length);
   };
 
   BN.prototype.toArray = function toArray (endian, length) {
+    return this.toArrayLike(Array, endian, length);
+  };
+
+  BN.prototype.toArrayLike = function toArray (Cons, endian, length) {
```

Comment:
`toArrayLike` in function's name ;) . Is there any point in making it public? If you think so - please add docs to readme